cluster/swarm/cluster.go

```diff
@@ -452,9 +452,7 @@ func (c *Cluster) Image(IDOrName string) *cluster.Image {
 		return nil
 	}
 
-	c.RLock()
-	defer c.RUnlock()
-	for _, e := range c.engines {
```

Comment:
Is the lock only there for engine iteration, or is there any worry of the image lookup failing due to an engine going away as well?